icu4c/source/test/intltest/dtfmttst.cpp

```diff
@@ -5555,6 +5556,44 @@ void DateFormatTest::TestAdoptCalendarLeak() {
     sdf.adoptCalendar(Calendar::createInstance(status));
 }
 
+/**
+ * Test that 'a' and 'B' fields are not duplicated in the field position iterator.
+ */
+void DateFormatTest::Test20741_ABFields() {
+    IcuTestErrorCode status(*this, "Test20741_ABFields");
+
+    const char16_t timeZone[] = u"PST8PDT";
+
+    UnicodeString skeleton = u"EEEEEBBBBB"; 
+    int32_t count = 0;
+    const Locale* locales = Locale::getAvailableLocales(count);
+    for (int32_t i = 0; i < count; i++) {
```

Comment:
Looping over all locales is too slow.  I just tested it.  We try to keep tests at no more than 100 ms (hopefully faster), and this takes 910 ms.

Please add:

```cpp
if (quick && (i % 17) != 0) { continue; }
```

to make the test run on 1/17th of locales, except in exhaustive mode when it still runs on all locales.